pkgs/by-name/au/autobrr/package.nix

```diff
@@ -0,0 +1,96 @@
+{ lib, cacert, esbuild, jq, moreutils, buildGoModule, fetchFromGitHub, stdenvNoCC, nodePackages, typescript }:
+
+let
+  pname = "autobrr";
+  version = "1.35.1";
+  src = fetchFromGitHub {
+    owner = "autobrr";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-pjr/Sb2rNZQfeATb58oTxj9l1SF8+VOZBMGUh1cBW+o=";
+  };
+
+  pnpm-deps = stdenvNoCC.mkDerivation {
+
+    pname = "${pname}-npm-deps";
+    inherit src version;
+
+    nativeBuildInputs = [
+      jq
+      moreutils
+      nodePackages.pnpm
+      cacert
+    ];
+
+    dontBuild = true; #this is just a download
+    dontCheck = true;
+    installPhase = ''
+      runHook preInstall
+
+      export HOME=$(mktemp -d)
+      pnpm config set store-dir $out
+      # use --ignore-script and --no-optional to avoid downloading binaries
+      # use --frozen-lockfile to avoid checking git deps
+      pnpm --dir web fetch
+
+      # Remove timestamp and sort the json files
+      rm -rf $out/v3/tmp
+      for f in $(find $out -name "*.json"); do
+        sed -i -E -e 's/"checkedAt":[0-9]+,//g' $f
+        jq --sort-keys . $f | sponge $f
+      done
+
+      runHook postInstall
+    '';
+
+    dontFixup = true;
+    outputHashMode = "recursive";
+    outputHash = "sha256-babKc60nuu+rfI5nnp7efvo4Z9Iz93CkH4qg7l3rRxA=";
+  };
```

Comment:
Until #231513 is resolved, we don't have a good user-story for `pnpm` in nixpkgs.

If upstream has a pre-built frontend packaged, I would suggest you use that for now.

Otherwise I would suggest manually generating a `package-lock.json`/`yarn.lock` and make use of `buildNpmPackage`/`mkYarnPackage` for now.

The reason being that this FOD will break when `pnpm` changes the format for its store...